File: testdata/bench/javascript/BenchmarkTest00006.js
// Safe: SQL injection prevented via parseInt validation
const express = require('express');
const db = require('./db');
const router = express.Router();
router.get('/test006', (req, res) => {
    const param = req.query.id;
    const safeId = parseInt(param, 10);
    if (isNaN(safeId)) { return res.status(400).json({ error: 'Invalid ID' }); }
    const sql = "SELECT * FROM users WHERE id = " + safeId;
    db.query(sql, (err, results) => { res.json(results); });
});
module.exports = router;

File: testdata/bench/rust/BenchmarkTest00006.rs
use actix_web::{web, HttpResponse};
use std::collections::HashMap;

// Safe: SQL injection prevented with diesel filter
async fn handler(query: web::Query<HashMap<String, String>>) -> HttpResponse {
    let param = query.get("name").unwrap_or(&String::new()).clone();
    // users::table.filter(users::name.eq(&param)).first::<User>(&conn)
    HttpResponse::Ok().body("safe")
}
